src/app/components/core/control-dispatcher/add-control-schedule-task/add-control-schedule-task.component.ts

```diff
@@ -0,0 +1,315 @@
+    // remove parameter from the list
+    const control = <FormArray>this.controlForm.controls['parameters'];
+    control.removeAt(index);
+    this.controlForm.markAsDirty();
+  }
+
+  clearForm() {
+    const control = <FormArray>this.controlForm.controls['parameters'];
+    control.clear();
+  }
+
+  getParametersFormControls(): AbstractControl[] {
+    return (<FormArray>this.controlForm.get('parameters')).controls
+  }
+
+  public toggleDropDown(id: string) {
+    const dropdowns = document.getElementsByClassName('dropdown');
+    for (let i = 0; i < dropdowns.length; i++) {
+      var openDropdown = dropdowns[i];
+      if (openDropdown.classList.contains('is-active')) {
+        openDropdown.classList.toggle('is-active', false);
+      } else {
+        if (openDropdown.id === id) {
+          openDropdown.classList.toggle('is-active');
+        }
+      }
+    }
+  }
+
+  public getScripts() {
+    this.controlService.fetchControlServiceScripts()
+      .subscribe((data: any) => {
+        this.scripts = data.scripts;
+      }, error => {
+        if (error.status === 0) {
+          console.log('service down ', error);
+        } else {
+          this.alertService.error(error.statusText);
+        }
+      });
+  }
+
+  refresh() {
+    this.clearForm();
+    this.getScriptByName();
+  }
+
+  getScriptByName() {
+    this.ngProgress.start();
+    this.controlService.fetchControlServiceScriptByName(this.script)
+      .subscribe((data: any) => {
+        this.scriptData = data;
+        this.ngProgress.done();
+        this.script = data.name;
+        this.getScriptParameters(data.configuration?.write?.value)
+      }, error => {
+        /** request completed */
+        this.ngProgress.done();
+        if (error.status === 0) {
+          console.log('service down ', error);
+        } else {
+          this.alertService.error(error.statusText);
+        }
+      });
+  }
+
+  getScriptParameters(value) {
+    if (value) {
+      let config = JSON.parse(value);
+      let params = config.map(c => c.values);
+      let parameters = [];
+      // generate parameters array
+      params.forEach(element => {
+        if (!element.hasOwnProperty("")) {
+          for (const [key, value] of Object.entries(element)) {
+            parameters.push({ key, value });
+          }
+        }
+      });
+      // remove duplicate objects from array
+      parameters = [...new Set(parameters.map(o => JSON.stringify(o)))].map(s => JSON.parse(s));
+      // generate parameters form controls
+      parameters.forEach(e => {
+        this.addParameter({ key: e.key, value: e.value });
+      });
+      return parameters;
+    }
+  }
+
+  setScript(script: any) {
+    this.scriptData = script;
+    // Get write steps from script
+    this.getControlParameters();
+    this.script = script.name;
+    this.controlForm.markAsDirty();
+  }
+
+  getControlParameters() {
+    this.clearForm();
+    this.writeScriptParameters = [];
```

Comment:
could you please file a Jira and also add other step types missing config for substitution.
.